src/canvas.py

```diff
@@ -273,6 +276,37 @@ def set_legend(self):
 
 class DummyToolbar(NavigationToolbar2):
     """Own implementation of NavigationToolbar2 for rubberband support."""
+    # Overwritten function - do not change name
+    def _zoom_pan_handler(self, event):
+        if event.button != 1:
+            return
+        if self.canvas.application.interaction_mode == InteractionMode.PAN:
+            if event.name == "button_press_event":
+                self.press_pan(event)
+            elif event.name == "button_release_event":
+                self.release_pan(event)
+        elif self.canvas.application.interaction_mode == InteractionMode.ZOOM:
+            if event.name == "button_press_event":
+                self.press_zoom(event)
+            elif event.name == "button_release_event":
+                self.release_zoom(event)
+
+    # Overwritten function - do not change name
+    def _update_cursor(self, event):
+        mode = self.canvas.application.interaction_mode
+        if event.inaxes and event.inaxes.get_navigate():
+            if (mode == InteractionMode.ZOOM
+                    and self._last_cursor != tools.Cursors.SELECT_REGION):
+                self.canvas.set_cursor(tools.Cursors.SELECT_REGION)
+                self._last_cursor = tools.Cursors.SELECT_REGION
+            elif (mode == InteractionMode.PAN
+                  and self._last_cursor != tools.Cursors.MOVE):
+                self.canvas.set_cursor(tools.Cursors.MOVE)
+                self._last_cursor = tools.Cursors.MOVE
+        elif self._last_cursor != tools.Cursors.POINTER:
+            self.canvas.set_cursor(tools.Cursors.POINTER)
+            self._last_cursor = tools.Cursors.POINTER
+
```

Comment:
What do these two overwritten function change compared to their original counterparts?

Reply:
The base implementation checks against the mode stored inside the toolbar itself. The difference here is, that we check against InteractionMode stored inside the main application, deduplicating logic